lib/cn/cn_tree.c

```diff
@@ -1204,13 +1202,11 @@ cn_tree_prepare_compaction(struct cn_compaction_work *w)
     merr_t err = 0;
     size_t outsz = 0;
     u32 n_outs;
-    u32 fanout;
     uint i;
     const bool kcompact = (w->cw_action == CN_ACTION_COMPACT_K);
     const bool split = (w->cw_action == CN_ACTION_SPLIT);
 
-    fanout = w->cw_tree->ct_fanout;
-    n_outs = fanout;
+    n_outs = CN_FANOUT_MAX;
```

Comment:
This needn't be any larger than the number of nodes in the tree.  And maybe for incremental spill 1?
About incremental spill, it might be more efficient to only spill what you have so far if there's someone waiting, otherwise maybe just keep going until done.

Reply:
I think @gsramdasi can change this to 2 with his PR.  Split will need to outputs, all others will need 1.  Spill needs CN_MAX_FANOUT because we currently hold on to all outputs until the rspill operation is complete.

Reply:
It's not clear how much waste/churn is happening if we have to allocate the max size thing for each spill.  When memory is tight allocating a 1M when we only need 16K is more expensive as it will have to got into the kernel and acquire the map lock and all that jazz, vs just grabbing a buffer from a malloc zone.  Or some such...